udata/core/user/api.py

```diff
@@ -238,14 +238,26 @@ def post(self):
         return user, 201
 
 
+def is_not_available(user):
+    return user.active is False and (
+        current_user.is_anonymous or
+        current_user.sysadmin is False)
+
+
 @ns.route('/<user:user>/', endpoint='user')
 @api.response(404, 'User not found')
-@api.response(410, 'User has been deleted')
+@api.response(410, 'User is not active or has been deleted')
 class UserAPI(API):
+
+    def check_availability(self, user):
```

Comment:
I agree, it smells like a decorator and that is actually what I started doing... before I stopped !

Indeed, if you put it as a decorator, it actually does not reduce the size of the code and it make it less readable.